indico/web/client/js/custom_elements/ind_date_picker.scss

```diff
@@ -0,0 +1,165 @@
+// This file is part of Indico.
+// Copyright (C) 2002 - 2024 CERN
+//
+// Indico is free software; you can redistribute it and/or
+// modify it under the terms of the MIT License; see the
+// LICENSE file for more details.
+
+@use 'design_system';
+@use 'partials/icons';
+
+ind-date-picker {
+  display: flex;
+  flex-direction: column;
+  gap: 0.2em;
+  align-items: flex-start;
+  position: relative;
+
+  > input {
+    padding-right: 2.5em;
+  }
+
+  > button {
+    @extend %text-size-reset;
+    @extend %button-icon-only;
+
+    position: absolute;
+    right: 0.2em;
+    top: 0.2em;
+    width: 2em;
+    aspect-ratio: 1;
+  }
+
+  > button::before {
+    @extend %icon;
+    @extend %icon-calendar;
+  }
+
+  > button span {
+    @extend %visually-hidden;
+  }
+}
+
+ind-calendar {
```

Comment:
While it's usually me who complains about things being too big and having too much margin/padding, this time I did not have that feeling most of the guys in the team find the date picker a bit too big:

- "Am I the only one who finds the font size/padding of the new datepicker way too big?"
- "This is massive"
- "it looks kinda huge"

Is there anything to tweak w/o reducing a11y? I think on mobile it has the right size (even though personally I usually prefer an `input[type=date]` on mobile, but that's off-topic here) since you can easily tap calendar cells there, but the calendar dialog is indeed quite large.

cc @tomasr8 @GovernmentPlates @duartegalvao

Reply:
The main issue with the size is actually that styling of the inputs propagates all the way down to the inputs used in the month-year controls. Basically semantic UI applies the input styling too broadly to anything that is inside `.ui.field` or whatever the class is. It can be made smaller if I get rid of the excessive paddings in the select list and text input, but that means more code whose only purpose is to deal with bad design of the Semantic UI. In these cases I generally prefer to err on the side of not having (too many) unnecessary overrides.

Reply:
Of course, I'm not strictly opposed to adding such overrides if that's what you prefer.

Reply:
Just thought I would chime in with my own two pence: I'm all for a11y, but as we have already said before, this widget is simply way too big in its current form (not to mention, the size makes this look out-of-place when comparing this to other picker elements we already have), and I personally would rather see this polished/fixed (i.e. reducing the visual size of this) before pushing this up into master.

I know SUI can be a pain in the neck sometimes (this is a recurring theme for all of us here :P - not to mention, other teams at CERN which use React SUI), but please consider fixing this, even if it means doing some annoying overrides.

Reply:
Sure, no problem.

There appears to be some misunderstanding here, though, so let me try to correct that first.

I agree that the widget is too large. I am also not opposed to fixing that. My point was simply that this adds more technical debt, and from what I see, you're already swimming in it. I just wanted confirmation that you're ok with adding more of it.

Reply:
I think we can't really avoid it... :/
So yeah, we're OK with it

Reply:
I think it's now reasonably sized. I can make it 12.5% smaller if you'd like, but that would negatively impact the click target size, so I wouldn't recommend that.

I thought a bit about how we can make overrides painfully clear, and decided to implement a few lightweight mixins that you can later hunt down if decide to address debt.

I've also run into this bit of CSS in `_input.scss`:

```css
  &:-moz-focusring {
    color: transparent !important;
    text-shadow: 0 0 0 $black !important;
  }
```

This generally breaks select lists when they're styled normally (outside of SUI). You might want to consider removing it, and apply proper styling for whatever it's trying to "fix".

Reply:
Looks pretty nice now IMHO. Just wondering: Can we easily make this a bit responsive to keep it large on small mobile screens? While it's much better now on desktops, on my phone the large size made it much easier to select a date.

Reply:
Let me see what I can do.

Reply:
Ok, so I'd leave this for some other time. Since Indico doesn't scale correctly on mobile devices, it would require some device detection, which just adds flakiness to the design. I don't think we should pursue that solution.

Reply:
Fine for me, it's already better than the old one anyway (tested w/ the accommodation field one).